### DISCUSSION TITLE
Problem with TWIG date filter?

### DISCUSSION THREAD

**Discussion by rbuehler-teletrend** [kimai/kimai#4696] [General] [answered]
Hi,

I'm having a strange behavior with using `invoice['query.begin']|date('d.m.Y')` in an invoice template.
For the same invoice date query range 01.02.2024-29.02.2024:
While `invoice['query.begin']` returns the correct date 01.02.24, `invoice['query.begin']|date('d.m.Y')` would always return the current date like 12.03.2024.

The problem doesn't exist for the end date:
`invoice['query.end']` returns 29.02.24
`invoice['query.end']|date('d.m.Y')` returns 29.02.2024

Any idea what's wrong? Is this a TWIG date filter bug?

**Comment by kevinpapst** [comment, marked as answer]
No, that is a "user is not reading the docs properly" bug 😁
<img width="1208" alt="Bildschirmfoto 2024-03-12 um 17 12 53" src="https://github.com/kimai/kimai/assets/533162/1d0fd42e-9d19-42b7-ac59-89bfeaebb648">

https://www.kimai.org/documentation/invoices.html

You are trying to use a string like "01.02.2024" as a date and then format it. PHP is clever and tries to convert it to a date, but in that case it fails, as it is ambigious. It could be second of january or first of february.
The other example works, because 29 cannot be a month.

Use ${query.begin_day} and alike.
